src/coords.c

```diff
@@ -238,19 +237,22 @@ npy_intp CountCoords(PyArrayObject *arr)
     if (iternext == NULL) { NpyIter_Deallocate(iter); return -1; }
     dataptr = NpyIter_GetDataPtrArray(iter);
 
+    GEOS_INIT;
+
     do {
         /* get the geometry */
         obj = *(GeometryObject **) dataptr[0];
-        if (!get_geom(obj, &geom)) { result = -1; goto finish; }
+        if (!get_geom(obj, &geom)) { errstate = PGERR_NOT_A_GEOMETRY; result = -1; goto finish; }
```

Comment:
Suggestion: here and line 250, can this get expanded to multiple lines for readability?